src/cmd/services/m3query/config/config.go

```diff
@@ -106,3 +110,27 @@ type RPCConfiguration struct {
 	// coordinator calls.
 	RemoteListenAddresses []string `yaml:"remoteListenAddresses"`
 }
+
+// TagOptionsConfiguration is the configuration for shared tag options
+// Currently only name, but can expand to cover deduplication settings, or other
+// relevant options.
+type TagOptionsConfiguration struct {
+	// MetricName specifies the tag name that corresponds to the metric's name tag
+	// Defaults to __name__ for Prometheus.
+	MetricName string `yaml:"metricName"`
+}
+
+// TagOptionsFromConfig translates tag option configuration into tag options.
+func TagOptionsFromConfig(cfg TagOptionsConfiguration) (models.TagOptions, error) {
+	opts := models.NewTagOptions()
+	name := cfg.MetricName
+	if name != "" {
+		opts = opts.SetMetricName([]byte(name))
+	}
+
+	if err := opts.Validate(); err != nil {
```

Comment:
I'm guessing this is for future validations?

Reply:
Yeah, using a similar pattern to how some of our other configs look, so in a similar vein to [the db client options](https://github.com/m3db/m3/blob/master/src/dbnode/client/types.go#L211)